### DISCUSSION TITLE
statistics map

### DISCUSSION THREAD

**Discussion by novw** [2Abendsegler/GClh#2009] [Ideas]
When in statistics -> map -> caches you've found, when hovering the map the country and the number of founds pops up. Why not make that clickable like the text links to the right of the map
![Naamloos](https://user-images.githubusercontent.com/84809678/149810373-cb367735-532e-42c3-a4df-26783a1f62f2.jpg)
?

**Comment by 2Abendsegler** [comment]
Hi,

at least at first glance, the development seems quite complex.
 
The added value does not seem particularly great to me because the functionality is almost identical in the list. You have many countries. And the more countries you have, the more you may have to scroll through the list. But that's not that much more complicated.

I don't want to implement the feature at the moment.

At the moment all other developers seem to be putting their feet up. This could be due to winter fatigue. So maybe there is no one there who wants to do the development.

But you're welcome to open an [issue](https://github.com/2Abendsegler/GClh/issues/new?assignees=&labels=enhancement&template=feature_request.yml) about it. Maybe there will be someone who wants to develop it. I will then prioritize the issue with very low. If it is not realized within a year, I will archive it. 

I hope this is ok.
Cheers,
Frank

**Comment by capoaira** [comment]
Hey,
I look at this, and I also think that this is quite complex, Maybe too complex compared to the added value.
At the moment I have not enough time to develop such a feature. But if somebody else wants to give a try, here are some information I found: In a `<g>` tag of the SVG is a `g.google-visualization-tooltip` insert on mouse over and removed on mouse leaf. The country code looks like the same as the country codes for urls (e.g. de, nl, se, ...) and are the same as the country code in the List. So, maybe, there is possibility to implement it.

**Comment by capoaira** [comment]
Hey,
after a quick look yesterday, I had an idea in the evening. It wasn't as complex as expected.
You can test it [here](https://github.com/2Abendsegler/GClh/blob/a1e405e05e224e7ce67ec31b644606ce016b0649/gc_little_helper_II.user.js?raw=true)

**Reply by novw** [reply]
Dear Capoaira. I love you for trying to implement this, but I have absolutely zero idea what I can do to help. If I click the link all I see is "All script settings will be reset!". This means I will loose all settings I made? It took me months (!) to adjust settings. One. By. One... I sure hate to loose that :( :( 

--
aha. I changed the "version" to 0.11.10 and manually saved the script. Disabled the 0.11.9 and I am trying now.

**Reply by 2Abendsegler** [reply]
> If I click the link all I see is "All script settings will be reset!". This means I will loose all settings I made?

This does not mean your GC little helper II settings, but the GC little helper II script settings in Tampermonkey. 

Regardless of this, you can save your GC little helper II settings. Maybe, take a look at your [dashboard](https://www.geocaching.com/account/dashboard). Below your user name there is the link "GClh II Sync". Or here the direct link to the [GClh II Sync](https://www.geocaching.com/account/dashboard#GClhShowSync). This allows you to save and load your GC little helper II settings in an DropBox account, export or import them manually, or download them directly as a file. 

If you have invested so much time, you should definitely do it. 😉

**Comment by novw** [comment]
It works great on country-maps (Caches You've Found in the World , Caches You've Found in Europe , etc), but not on states, provinces or other maps. And personally (! thats just me) I think a "click for filter country" text might be appropriate? 

Also: the target="_blank" seems to fail. The filtered list is opened in the same tab as the map. (both the map as the sidelist)

If this was "so easy" (lol), is it possible to make the UNfound countries pop up which country they are? I mean, I can hover over a "yellow" country, but am (sometimes) completely clueless about what country I am hovering over. The map "knows" I am hovering a country (borders thicken), it's just me not knowing my Kazakhstan from my Kirghistan or Turkmenistan :)

**Reply by capoaira** [reply]
>It works great on country-maps (Caches You've Found in the World , Caches You've Found in Europe , etc), but not on states, provinces or other maps. 

I only have the world and Europe map. Can you tell me your geocaching name, then I can try to test the feature with your profil. 

> And personally (! thats just me) I think a "click for filter country" text might be appropriate?

Yes, but text on hover (like on links) are for svgs not as easy as for links. To insert it into the existing hover-text looks not good, I think.

> Also: the target="_blank" seems to fail. The filtered list is opened in the same tab as the map. (both the map as the sidelist)

I agree with that. Schouldn't be a problem to implement this.

> is it possible to make the UNfound countries pop up which country they are?

Unfortunately, that is impossible. The data comes from a GS script that we do not have access to. We only find out which country is being hovered over from the visible popup.

**Reply by novw** [reply]
When hovering on a country with finds, there is a popup (country code + number of finds). Maybe in that popup is room for "click for more"? (align=right after the country letters. Use the fact there is no country name in the popup, just a code. The popup is large enough I think). Otherwise I doubt many people will even know the map is now clickable. Or add a line of text above the map: "click country with finds to filter"?

My caching name is "novw.nl".

Public statement : I appreciate what you are doing! Thank you

**Reply by capoaira** [reply]
Now it's as complex as expected :(
I could not find any given association between state codes and the states in the list. For this reason, the whole thing is very complex, and I have to put it on hold. :(

**Reply by 2Abendsegler** [reply]
> I could not find any given association between state codes and the states in the list.

I couldn't find anything either. 
But we should deliver the actual enhancement with the next release, right?

**Reply by capoaira** [reply]
Hm, its only working for country maps, but not for states. Thats can be confusing if one map work, but an other not.

**Reply by 2Abendsegler** [reply]
Yes, you are right.

**Reply by novw** [reply]
Sad (: but I understand. Thanks for trying!

**Comment by 2Abendsegler** [comment]
@novw 
We have found solutions for some of the features and have implemented them in the new version that is just out.
This is just for your information.
